src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.csproj

```diff
@@ -445,24 +447,10 @@
     <ProjectReference Include="..\..\external\Java.Interop\src\Java.Interop\Java.Interop.csproj" />
   </ItemGroup>
 
-  <ItemGroup>
-    <Compile Update="Properties\Resources.Designer.cs">
-      <DesignTime>True</DesignTime>
-      <AutoGen>True</AutoGen>
-      <DependentUpon>Resources.resx</DependentUpon>
-    </Compile>
-  </ItemGroup>
-
-  <ItemGroup>
-    <EmbeddedResource Update="Properties\Resources.resx">
-      <Generator>ResXFileCodeGenerator</Generator>
-      <LastGenOutput>Resources.Designer.cs</LastGenOutput>
-    </EmbeddedResource>
-  </ItemGroup>
-
   <!-- Import Microsoft.NET.Sdk targets before our targets so we can override behavior -->
   <!-- See https://github.com/microsoft/msbuild/pull/4922 -->
   <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" />
+  <Import Project="..\tools\GenerateResxSource.targets" Sdk="Microsoft.DotNet.Arcade.Sdk" />
```

Comment:
In theory, this project could also now import `..\tools\Localization.targets` instead of using an explicit `<PackageReference/>` for XliffTasks, but `..\tools\Localization.targets` makes some additional assumptions about other properties that need to be set. One way to split the difference would be to import just `..\tools\DefaultVersions.Generated.props` to get `$(XliffTasksVersion)` from the Microsoft.DotNet.Arcade.Sdk NuGet package, but even then, since the change isn't specifically required for this PR, it's probably cleanest to consider it in a separate PR.

Reply:
Is there documentation on what files you can import like this? It seems to me that the names of these files might change between versions of Microsoft.DotNet.Arcade.Sdk, unless they expect people to import them directly.

They have a lot of documentation here, it just doesn't seem to be organized too well:

https://github.com/dotnet/arcade/tree/master/Documentation

Reply:
Ok, I played around with:
```xml
<Import Project="Sdk.props" Sdk="Microsoft.DotNet.Arcade.Sdk" />
...
<Import Project="Sdk.targets" Sdk="Microsoft.DotNet.Arcade.Sdk" />
```
It seems like this Sdk does a lot more things than we need:
* computes many types of version numbers
* signs with a Microsoft .snk
* generates `[assembly:InternalsVisibleTo]`
* sets up `Test` targets

Using this diff worked, but I don't think we should do this:

```diff
--- a/src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.csproj
+++ b/src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.csproj
@@ -1,7 +1,7 @@
 <Project>

   <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
-  <Import Project="..\tools\BuildTasks.props" Sdk="Microsoft.DotNet.Arcade.Sdk" />
+  <Import Project="Sdk.props" Sdk="Microsoft.DotNet.Arcade.Sdk" />
   <Import Project="..\..\Configuration.props" />
   <Import Project="..\..\build-tools\scripts\LocalizationLanguages.projitems" />
   <Import Project="..\..\build-tools\scripts\MSBuildReferences.projitems" />
@@ -19,6 +19,7 @@
     <LangVersion>latest</LangVersion>
     <NoWarn>8632</NoWarn>
     <GenerateResxSource>true</GenerateResxSource>
+    <SignAssembly>false</SignAssembly>
   </PropertyGroup>

   <ItemGroup>
@@ -450,7 +451,7 @@
   <!-- Import Microsoft.NET.Sdk targets before our targets so we can override behavior -->
   <!-- See https://github.com/microsoft/msbuild/pull/4922 -->
   <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" />
-  <Import Project="..\tools\GenerateResxSource.targets" Sdk="Microsoft.DotNet.Arcade.Sdk" />
+  <Import Project="Sdk.targets" Sdk="Microsoft.DotNet.Arcade.Sdk" />
   <Import Project="Xamarin.Android.Build.Tasks.targets" />
```

These attributes got added:
```
[assembly: AssemblyInformationalVersion("1.0.0-dev")]
[assembly: AssemblyCopyright("© Microsoft Corporation. All rights reserved.")]
[assembly: AssemblyFileVersion("42.42.42.42424")]
[assembly: AssemblyVersion("42.42.42.42")]
```
Some obsession with 42 here?

Is there another team using this for localization-only? I looked at dotnet/sdk, and they use it to do all of the above.

If we want to use this, I think what you did is the best option, since we don't want all this other stuff in here.

Reply:
42 is the answer is it not?  :P

I don't like the idea of doing all that extra stuff. Having looked at the code, there appears no way to turn bits off via properties if you import the `Sdk.targets`.

Reply:
Just to throw out one other idea, it looks like another option could instead be to copy just the one file for the desired task ([_GenerateResxSource.cs_](https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/src/GenerateResxSource.cs)) into xamarin-android-tools (so it can be used by all the Xamarin.Android repos), probably also along with the _.targets_ file. Based on a hunch, I tried a quick experiment of copying that single _GenerateResxSource.cs_ file into an empty C# project, and it did build, so it looks like it doesn't currently depend on anything else in Arcade.

At least one disadvantage would be that any potential fixes from upstream would need to be copied by hand into xamarin-android-tools instead of just updating the NuGet package. (There might also be a few details to work out about how to consume the task from xamarin-android-tools via the build process of each repository.) On the other hand, it would provide some more transparency in cases where an API change or file rename in new Arcade versions might break xamarin-android's direct import of _GenerateResxSource.targets_. If there's precedent for preferring the single-file approach, maybe it'd be an acceptable option here too?

Reply:
I just realized the single-file approach would be messy in the case of LibZipSharp because LibZipSharp would also need its own separate copy of the task and the targets, unless it added a submodule dependency on xamarin-android-tools, and adding that dependency doesn't seem right for LibZipSharp because that library is not specific to Xamarin.Android. So I'm thinking that copying the single file around is probably uglier overall than importing the specific files from the NuGet package.

If we wanted to either use the current approach of this PR temporarily or abandon it for now but try to come back to using `GenerateResxSource` again in the future, one possible option might be to see if the the Arcade maintainers would be OK with splitting that task into its own NuGet package eventually, or see if they'd be OK documenting API stability for `..\tools\GenerateResxSource.targets`.